src/com/sun/jna/LastErrorException.java

```diff
@@ -44,23 +50,17 @@ public LastErrorException(String msg) {
                 msg = msg.substring(1, msg.indexOf("]"));
             }
             this.errorCode = Integer.parseInt(msg);
-        }
-        catch(NumberFormatException e) {
+        } catch(NumberFormatException e) {
             this.errorCode = -1;
         }
     }
 
-    /**
-     * Returns the error code of the error.
-     * @return
-     *  Error code.
-     */
-    public int getErrorCode() {
-    	return errorCode;
+    public LastErrorException(int code) {
+        this(code, formatMessage(code));
     }
 
-    public LastErrorException(int code) {
-        super(formatMessage(code));
+    protected LastErrorException(int code, String msg) {
```

Comment:
Not really, I just thought it would discourage programmers from creating a new instance that contains code _and_ message unless they are using a derived class to do so (e.g., _LinuxException_, _MacException - similar to _Win32Exception_). However,I am quite fine with making it _public_...